src/protocol/attribution/aggregate_credit.rs

```diff
@@ -100,33 +261,35 @@ async fn bit_decompose_breakdown_key<F: Field>(
 }
 
 /// Sort the input by `aggregation_bit` first, then by `breakdown_key`
-#[allow(dead_code)]
 async fn sort_by_aggregation_bit_and_breakdown_key<F: Field>(
     ctx: SemiHonestContext<'_, F>,
     input: &[CappedCreditsWithAggregationBit<F>],
 ) -> Result<Vec<CappedCreditsWithAggregationBit<F>>, Error> {
-    // Sort by aggregation_bit
-    let sorted_by_aggregation_bit = sort_by_aggregation_bit(ctx.clone(), input).await?;
+    // Prepend the aggregation_bit bits with breakdown_key bits. This results
+    // in sorting the sidecar by aggregation_bit first, and then by breakdown_key.
+
+    let mut permutation_keys = vec![input
+        .iter()
+        .map(|x| x.aggregation_bit.clone())
+        .collect::<Vec<_>>()];
```

Comment:
Yeah, none of this should be necessary. We can construct the list in such a way that it is already sorted.